source/extensions/filters/common/ext_authz/ext_authz.h

```diff
@@ -68,16 +68,23 @@ struct Response {
   // A set of HTTP headers returned by the authorization server, that will be optionally appended
   // to the request to the upstream server.
   Http::HeaderVector headers_to_append;
+  // A set of HTTP headers returned by the authorization server, that will be optionally be added
+  // to the request to the upstream server if they don't already exist.
+  Http::HeaderVector headers_to_add_if_absent;
   // A set of HTTP headers returned by the authorization server, will be optionally set
   // (using "setCopy") to the request to the upstream server.
   Http::HeaderVector headers_to_set;
   // A set of HTTP headers returned by the authorization server, will be optionally added
   // (using "addCopy") to the request to the upstream server.
   Http::HeaderVector headers_to_add;
+  // A set of HTTP headers returned by the authorization server, will be optionally set
+  // (using "setCopy") to the response sent back to the downstream client (if they are absent)
+  // on OK auth responses.
+  Http::HeaderVector response_headers_to_add_if_absent;
```

Comment:
To add color to the previous comment, this new field to conditionally remove headers is in fact appropriate since the ext_authz filter can't know ahead of time what the upstream response headers will be.

Reply:
Make sense. Will drop it as suggested. Thanks!